sdk/examples/xo_python/sawtooth_xo/xo_client.py

```diff
@@ -140,13 +163,37 @@ def _send_xo_txn(self, name, action, space=""):
         )
 
         batch_list = self._create_batch_list([transaction])
-
-        result = self._send_request(
-            "batches", batch_list.SerializeToString(),
-            'application/octet-stream'
-        )
-
-        return result
+        batch_ids = [b.header_signature for b in batch_list.batches]
+
+        if wait and wait > 0:
+            wait_time = 0
+            start_time = time.time()
+            self._send_request(
+                "batches", batch_list.SerializeToString(),
```

Comment:
This request should be sent with the wait parameter. The simplest way to do this is swap `"batches"` for `"batches?wait={}".format(wait)`.

A somewhat more robust implementation would be to refactor `XoClient._send_request` to take a dict of query parameters (or just arbitrary `**kwargs`), and then parse those into the url properly (`?` before the first query, `&` between subsequent queries, `=` between key and value).

If you _don't_ send this request with a wait query, you will get a response immediately, and you'll be hammering the REST API every 0.2 seconds.